File: python/Milky Doughnuts/weekend_snacks.py
def count_even(numbers):
    count = 0
    for number in numbers:
        if number % 2 == 0:
            count += 1
    return count


def linear_search(numbers, target):
    for index in  range(len(numbers)):
        if numbers[index] == target:
            return index
    return -1


def squares(number):
    perfect_square = []
    for index in range(1, number+1):
        perfect_square.append(index * index)
    return perfect_square



def array_equal(first_list, second_list):
    if len(first_list) != len(second_list):
        return False
    for index in range(len(first_list)):
        if first_list[index] != second_list[index]:
            return False
    return True

def remove_duplicates(numbers):
    result = []
    for number in numbers:
        if number not in result:
            result.append(number)
    return result

def flatten(arrays):
    result = []
    for row in arrays:
        for number in row:
            result.append(number)
    return result

def intersect_of(first_list, second_list): 
    result = []
    for number in first_list:
        if number in  second_list and number not in result:
            result.append(number)
    return result

File: java/Milky Doughnuts/WeekendSnacks.java
public class WeekendSnacks{
    public int countEven(int[] numbers){
        int count = 0;
        for(int number: numbers){
            if (number % 2 == 0)
                count++;
        }

         return count;
    

    }

    public int linearSearch(int[] numbers, int target) {
        for (int index = 0; index < numbers.length; index++){

            if(numbers[index] == target){
              return index;  
                }
        

        }
        return -1;
}
    public int[] squares(int number){
         int[] perfectSquare = new int [number];

        for(int index = 0; index < number; index++){
            perfectSquare[index] = (index + 1) * (index + 1);
        

        }
        return perfectSquare;
    }

    
    public boolean arrayEquals(int[] firstArray, int[] secondArray){
        if (firstArray.length != secondArray.length){
         return false;
            }
        for (int index = 0; index < firstArray.length; index++){
            if (firstArray[index] != secondArray[index]){
                    return false;
            }


        }
        return true;

        }

    public int[] flatten(int[][] arrays){
        int total = 0;
        for(int[] row : arrays) {
            total += row.length;
                }

        int[] result = new int[total];
        int index = 0;
        for(int[] row :arrays){
            for(int number : row){
                result[index++] = number;

            }

        }
        return result;
    }




}
